src/runtime/HalideRuntimeHexagonDma.h

```diff
@@ -0,0 +1,94 @@
+/**
+ *  This file contains the necessary APIs and Structures for Initiating
+ * and Terminating the Hexagon DMA transfer from Halide user space
+ * The Header is available to the user to pass the necessary details
+ * of the Frame and its type so that Halide issue a matching call to DMA driver
+ * This File also have APIs to create the DMA context for the First frame and Delete
+ * the DMA context for the Last frame
+ */
+#ifndef HALIDE_HALIDERUNTIMEHEXAGONDMA_H
+#define HALIDE_HALIDERUNTIMEHEXAGONDMA_H
+
+#include "HalideRuntime.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * User Format currently supported
+ */
+typedef enum halide_hexagon_dma_user_fmt {
+    NV12,
+    UBWC_NV12,
+    P010,
+    TP10,
+    NV124R,
+    UBWC_NV124R,
+} halide_hexagon_dma_user_fmt_t;
+
+/**
+ * Component of the frame
+ */
+typedef enum halide_hexagon_user_component {
+    LUMA_COMPONENT,                //frame has only Luma component
+    CHROMA_COMPONENT,              //frame has only Chroma Component
+    BOTH_LUMA_CHROMA               //frame has both Luma nad chroma components 
+} halide_hexagon_dma_user_component_t;
+
+/** file
+ *  Routines specific to the Halide Hexagon dma runtime.
+ */
+typedef int halide_hexagon_dma_handle_t;
+
+/**
+ * halide_hexagon_dma_device_interface
+ * Returns hexagon dma interface  */
+extern const struct halide_device_interface_t *halide_hexagon_dma_device_interface();
+
+/**
+ * halide_hexagon_dmart_wrap_buffer
+ * in: halide_buffer_t*
+ * in: unsigned char
+ * out: int
+ */
+extern int halide_hexagon_dmart_wrap_buffer(void *user_context, halide_buffer_t* buf, void *inframe);
+
+/**
+ * halide_hexagon_dmart_release_wrapper
+ * in: halide_buffer_t*
+ * out: int
+ */
+extern int halide_hexagon_dmart_release_wrapper(void *user_context, halide_buffer_t* buf);
+
+/**
+ * halide_hexagon_dmart_get_memory
+ * in: halide_buffer
+ */
+void* halide_hexagon_dmart_get_memory(void* user_context, halide_buffer_t *roi_buf);
+
+/**
+ * halide_buffer_copy
+ * in:void* user_Context
+ * out:halide_buffer_t src
+ * out:halide_buffer_t dest
+ * void* ptr */
+int halide_buffer_copy(void *user_context, halide_buffer_t *src_buf, void *ptr, halide_buffer_t *dst_buf);
+
+/**
+ * halide_hexagon_dmaapp_create_context
+ * CreateContext check if DMA Engines are available
```

Comment:
Most of these comment blocks for the prototypes seem redundant, they just take up space and repeat what is already apparent in the prototype. I would suggest replacing the content with a meaningful statement about the function, or removing it all together.